Add search by name feature, Add filter command, fix searching for thumbs

Waterdragen opened the pull request:
- command: `!cmini search [column] --name [name]`

![Screenshot (61)](https://github.com/Apsu/cmini/assets/96000090/35d93a68-a177-4ad4-bb74-057f8b4cd6ee)
_example of searching by column then filter the results by name_ 

![Screenshot (60)](https://github.com/Apsu/cmini/assets/96000090/1860c403-1954-4787-bd72-a60eb52e78e7)
_example of only searching by name_

- fix cases like `cmd --kwarg` where `--kwarg` is considered part of `args` (--kwargs happens to be a valid `cmd_kwarg`)
  - now returns `args` == "" instead of including the first kwarg
- refactored code to make it more concise

Waterdragen committed:
Fix and refactor `get_kwargs`

- fix cases like `cmd --kwarg` where `--kwarg` is considered part of `args` (--kwargs happens to be a valid `cmd_kwarg`)
  - now returns `args` == "" instead of including the first kwarg
- refactored code to make it more concise

Waterdragen committed:
[feat] Add search by name feature

usage: `!cmini [column] --name <name>`
column is optional

Waterdragen committed:
Fix and refactor `get_kwargs`

Waterdragen committed:
Fix and refactor `get_kwargs`

Fix and refactor `get_kwargs`

drowningnewt reviewed (changes requested):
FINGER_ALIASES
Change `'thumb': {'LT', "RT'"},` to `'thumb': {'LT', 'RT', 'TB'},` to include all thumb layouts
Add rt alias to search all right thumbs on layouts via `'rt': {'RT', 'TB'},`
Assuming behavior is desired, LT should be added to 'lh' and RT/TB should be added to 'rh'

kwargs
Add bool tb to make searching all right thumb layouts possible

Waterdragen committed:
Fix args type checking logic

Waterdragen committed:
Fix searching for thumbs

merged with https://github.com/Apsu/cmini/pull/36

Waterdragen committed:
Fix join strings with space

Waterdragen committed:
Add filter command

usage: `!cmini filter [--kwargs]`
Supported options:
-  `[--column <sfb/column, ...fingers>]`
- `[--homerow <keys / "sequence">]`
- `[--sort <metric>]`
- `[--name <name>]`
- `[--<metric> {< or >}{num}]`
- `[--partial]`
- `metrics: sfb, sfs, alt, red, roll, oneh, inroll, outroll, rolltal, inrolltal`

filter by column/homerow/metric, sort by metric / random then by name, search all layouts (partial) / only those with a-z keys

Waterdragen committed:
Merge pull request #2 from Waterdragen/Waterdragen-filter-command

Add filter command

Waterdragen commented:
command: `!cmini filter [--kwargs]`

Options:
-  `[--column <sfb/column, ...fingers>]`
- `[--homerow <keys / "sequence">]`
- `[--sort <metric>]`
- `[--name <name>]`
- `[--<metric> {< or >}{num}]`
- `[--partial]`
- `metrics: sfb, sfs, alt, red, roll, oneh, inroll, outroll, rolltal, inrolltal`

Summary:
- filter by column/homerow/metric
- sort by metric / random then by name
- toggle search all layouts (partial option) / only those with a-z keys

![Screenshot (62)](https://github.com/Apsu/cmini/assets/96000090/91deaca3-5aa8-450a-b147-ca607a2d3e83)
^ _example of chaining multiple metric filters, then sort by metric_

![Screenshot (63)](https://github.com/Apsu/cmini/assets/96000090/bb0c4dff-4d7f-417c-90e2-dcea22ce7b80)
^ _example of filtering by name_

![Screenshot (64)](https://github.com/Apsu/cmini/assets/96000090/ab33e5af-1862-4631-927c-b83444ee8975)
^ _example of filtering by column and homerow_

![Screenshot (65)](https://github.com/Apsu/cmini/assets/96000090/12ef74f0-675a-4432-9e3b-91adcd200a8b)
^ _example of disabling partial layouts (top, default) and enabling them (bottom)_
